platform/src/main/java/org/hillview/storage/CassandraDatabase.java

```diff
@@ -0,0 +1,264 @@
+                }
+            } catch (Exception e) {
+                throw new RuntimeException("Failed to parse token range", e);
+            }
+        }
+
+        public String toString() {
+            return "start: " + this.start_token + ", end: " + this.end_token +
+                ", endpoints: " + this.endpoints.toString();
+        }
+    }
+
+    /**
+     * Shows key-range (table partition) and endpoints (replica) which stores each key-range.
+     * This key-range distribution is the same on every node in the cluster.
+     */
+    private void setTokenRanges() {
+        String keyspace = this.info.database;
+        try {
+            TokenRange tr;
+            for (String tokenRangeString : this.probe.describeRing(keyspace)) {
+                tr = new TokenRange(tokenRangeString);
+                this.tokenRanges.add(tr);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static class CassTable {
+        public String keyspace;
+        public List<String> tables;
+
+        public CassTable(String keyspace, List<String> tables) {
+            this.keyspace = keyspace;
+            this.tables = tables;
+        }
+
+        public String toString(){
+            String result = this.keyspace;
+            for (String table : this.tables) {
+                result += "\n  " + table;
+            }
+            return result;
+        }
+    }
+
+    /** The list of keyspaces and tables are obtained from cluster connection's metadata */
+    private void setStoredTableInfo() {
+        List<KeyspaceMetadata> keyspaces = this.metadata.getKeyspaces();
+        String keyspace;
+        Collection<TableMetadata> tables;
+        List<String> tableNames;
+        for (KeyspaceMetadata kMetadata : keyspaces) {
+            keyspace = kMetadata.getName();
+            tables = this.metadata.getKeyspace(keyspace).getTables();
+            tableNames = tables.stream().map(tm -> tm.getName())
+                .collect(Collectors.toList());
+            this.cassTables.add(new CassTable(keyspace, tableNames));
+        }
+    }
+
+    /** Returns list of tables stored in the cluster  */
+    public List<CassTable> getStoredTableInfo() {
+        return this.cassTables;
+    }
+
+    /** Given the keyspace and table name, sstableutil will return the sstable-related files.
+     * Using sstableutil command is better than doing manually sstable search. Although we know
+     * the sstable parent dir from cassandra.yaml, we need to rule out the outdated sstables.
+     * And sstableutil is built specifically to identify which version are the most updated.
+      */
+    public String getSSTablePath() {
+        String sstablePath = EMPTY;
+        boolean isWindows = System.getProperty("os.name")
+            .toLowerCase().startsWith("windows");
+        ProcessBuilder builder = new ProcessBuilder();
+        if (isWindows) {
```

Comment:
how about you don't use a shell, just call that command, and instead implement the grep in java on the standard output? Then you don't care whether this is windows or linux, it's the same command.

Reply:
Paths to create the path, then you don't care whether it's "/" or "\".